cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/aggregation/AggregationDefinition.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.aggregation;
+
+import io.cdap.cdap.api.annotation.Beta;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.relational.Expression;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Specifies how an aggregation should be executed.
+ */
+@Beta
+public abstract class AggregationDefinition {
+  private final List<Expression> groupByFields;
+  private final List<Expression> selectedFields;
+
+  protected AggregationDefinition() {
+    groupByFields = Collections.emptyList();
+    selectedFields = Collections.emptyList();
+  }
+
+  protected AggregationDefinition(List<Expression> groupByFields, List<Expression> selectedFields) {
+    this.groupByFields = Collections.unmodifiableList(groupByFields);
+    this.selectedFields = Collections.unmodifiableList(selectedFields);
+  }
+
+  public List<Expression> getGroupByFields() {
```

Comment:
Please add javadocs to all public / protected methods

Reply:
Done, let me know if the javadoc looks good and if I missed anything.